src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -970,6 +979,407 @@ public void recursivelyDelete(final Project project, final boolean commitIndex)
+            UUID uuid = it.next();
+            final Project project = getObjectByUuid(Project.class, uuid);
+            if (project == null) {
+                errorByUUID.put(String.valueOf(uuid), "Project not found");
+                it.remove();
+            }
+            else {
+                if (!hasAccess(principal, project)) {
+                    errorByUUID.put(String.valueOf(uuid), "Access denied to project");
+                    it.remove();
+                }
+            }
+        }
+        if (!errorByUUID.isEmpty()) {
+            throw ProjectOperationException.forDeletion(errorByUUID);
+        }
+
+        String[] uuidsArray = uuids.stream().map(UUID::toString).toArray(String[]::new);
+        String commaSeparatedUUIDs = String.join(",", Collections.nCopies(uuidsArray.length, "?"));
+        var queryParameter = Config.getInstance().getProperty(Config.AlpineKey.DATABASE_DRIVER).equals("com.microsoft.sqlserver.jdbc.SQLServerDriver") ?
+                commaSeparatedUUIDs : uuidsArray;
+        String inExpression = Config.getInstance().getProperty(Config.AlpineKey.DATABASE_DRIVER).equals("com.microsoft.sqlserver.jdbc.SQLServerDriver") ?
+            "IN(SELECT value FROM STRING_SPLIT(?, ','))" : "= ANY(?)";
+
+        runInTransaction(() -> {
+            try (var ignored = new ScopedCustomization(pm).withProperty(PROPERTY_QUERY_SQL_ALLOWALL, "true")) {
+                Query<?> sqlQuery = pm.newQuery(JDOQuery.SQL_QUERY_LANGUAGE, """
+                    DELETE FROM "PROJECTMETRICS" WHERE "PROJECT_ID" IN (
+                        SELECT "ID" FROM "PROJECT" WHERE "UUID" = ANY(?)
+                    )
+                    """.replace("= ANY(?)", inExpression)
+                );
+                executeAndCloseWithArray(sqlQuery, queryParameter);
+
+                sqlQuery = pm.newQuery(JDOQuery.SQL_QUERY_LANGUAGE, """                        
+                    DELETE FROM "DEPENDENCYMETRICS" WHERE "COMPONENT_ID" IN (
+                        SELECT "ID" FROM "COMPONENT" WHERE "PROJECT_ID" IN (
+                            SELECT "ID" FROM "PROJECT" WHERE "UUID" = ANY(?)
+                        )
+                    ) OR "PROJECT_ID" IN (
+                        SELECT "ID" FROM "PROJECT" WHERE "UUID" = ANY(?)
+                    );
+                    """.replaceAll(Pattern.quote("= ANY(?)"), inExpression)
+                );
+                executeAndCloseWithArray(sqlQuery, queryParameter, queryParameter);
+
+                sqlQuery = pm.newQuery(JDOQuery.SQL_QUERY_LANGUAGE, """                        
+                    DELETE FROM "FINDINGATTRIBUTION" WHERE "COMPONENT_ID" IN (
+                        SELECT "ID" FROM "COMPONENT" WHERE "PROJECT_ID" IN (
+                            SELECT "ID" FROM "PROJECT" WHERE "UUID" = ANY(?)
+                        )
+                    ) AND "PROJECT_ID" IN (
+                        SELECT "ID" FROM "PROJECT" WHERE "UUID" = ANY(?)
+                    );
```

Comment:
The condition on `PROJECT_ID` should be sufficient, as that is always populated.